design/scheduler.md

```diff
@@ -0,0 +1,158 @@
+# Scheduler design
+
+This document covers the design and implementation details of the swarmkit
+scheduler.
+
+## Overview
+
+The scheduler becomes responsible for tasks once they reach the `Pending` state.
```

Comment:
https://github.com/docker/swarmkit/blob/master/design/task_model.md#task-lifecycle covers that - the link to that specific subsection should be here instead of further below?